distribution/openrct2.d.ts

```diff
@@ -387,96 +585,211 @@ declare global {
+         * 
+         * @return The ride. TODO what if doesn't exist?
+         */
         getRide(id: number): Ride;
+
+        /**
+         * Get the tile at the given position. TODO what if OOB?
+         * 
+         * @param x The x-coordinate of the tile.
+         * @param y The y-coordinate of the tile.
+         * 
+         * @return The game tile. TODO what if OOB?
+         */
         getTile(x: number, y: number): Tile;
+
+        /**
+         * Get the object with the given id. TODO what if doesn't exist?
+         * 
+         * @param id The object id.
+         * 
+         * @return The object. TODO what if doesn't exist?
+         */
         getEntity(id: number): Entity;
+
+        /**
+         * Get all map objects of the given type.
+         * 
+         * @param type The object type.
+         * 
+         * @return An array of objects.
+         */
         getAllEntities(type: EntityType): Entity[];
+
+        /**
+         * Get all peep objects.
+         * 
+         * @param type The "peep" object type.
+         * 
+         * @return An array of Peeps.
+         */
         getAllEntities(type: "peep"): Peep[];
     }
 
+    /** Type of tile elements (e.g. surface, footpath, track). */
     type TileElementType =
         "surface" | "footpath" | "track" | "small_scenery" | "wall" | "entrance" | "large_scenery" | "banner"
-        /** This only exist to retrieve the types for existing corrupt elements. For hiding elements, use the isHidden field instead. */
+        /** This only exists to retrieve the types for existing corrupt elements. For hiding elements, use the isHidden field instead. */
         | "openrct2_corrupt_deprecated";
 
+    /**
+     * A tile element.
+     */
     interface BaseTileElement {
+        /** The tile type. */
         type: TileElementType;
+        /** The base height of the tile for collision. */
         baseHeight: number;
+        /** The top height of the tile for collision. */
         clearanceHeight: number;
-        isHidden: boolean; /** Take caution when changing this field, it may invalidate TileElements you have stored in your script. */
+        /** Indicates if the tile is hidden. Take caution when changing this field, it may invalidate 
+         * TileElements you have stored in your plugin. 
+         */
+        isHidden: boolean;
     }
 
+    /**
+     * A surface element.
+     */
     interface SurfaceElement extends BaseTileElement {
+        /** TODO what does the number mean? */
         slope: number;
+        /** A number indicating the surface style. */
         surfaceStyle: number;
+        /** A number indicating the edge style (blended or not?) TODO: verify */
         edgeStyle: number;
+        /** The surface height of the water. */
         waterHeight: number;
+        /** The length of the grass from X to Y. TODO */
         grassLength: number;
+        /** A number from x to y indicating the ownership status. TODO */
```

Comment:
In C++ this is in terms of flags @IntelOrca to confirm how this works in the plugins

Reply:
Yeah I haven't changed the field from C++.